pkg/apis/application/v1alpha1/types.go

```diff
@@ -2815,12 +2815,21 @@ func (source *ApplicationSource) Equals(other *ApplicationSource) bool {
 	if !source.Plugin.Equals(other.Plugin) {
 		return false
 	}
+	// helm source can't be compared directly because raw value sometime
+	// get html escaped and creating a diff
+	if source.Helm != nil && other.Helm != nil {
+		if source.Helm.ValuesString() != other.Helm.ValuesString() {
+			return false
+		}
```

Comment:
This only compares the ValueString for equality. You will need to create an Equality function like https://github.com/argoproj/argo-cd/blob/f89910ae58b469a096f92f3615e76780114ac6d4/pkg/apis/application/v1alpha1/types.go#L875, that compares the ValuesString and all other fields not included in ValueString.